Memorize/MemorizeGameModel.swift

```diff
@@ -0,0 +1,121 @@
+//
+//  MemorizeGameModel.swift
+//  Memorize
+//
+//  Created by Admin on 10.02.2023.
+//
+
+import Foundation
+
+struct MemorizeGame<CardContent> where CardContent: Equatable{
+    
+    private(set) var cards: Array<Card>
+    
+    private var indexOfOneFaceUpCard: Int?
+    
+    mutating func choose(_ card: Card) {
+        if let chosenIndex = cards.firstIndex(where: { $0.id == card.id}),
```

Comment:
please try using `guard` whenever it's possible. it's more convenient and code becomes more readable